internal/pubsub/pubsub.go

```diff
@@ -202,17 +188,14 @@ func (s *Subscription) Close() error {
 // It is for use by provider implementations.
 func NewSubscription(ctx context.Context, d driver.Subscription) *Subscription {
 	handler := func(item interface{}) {
-		mecs := item.([]msgErrChan)
+		ms := item.([]*Message)
 		var ids []driver.AckID
-		for _, mec := range mecs {
-			m := mec.msg
+		for _, m := range ms {
 			id := m.ackID
 			ids = append(ids, id)
 		}
-		err := d.SendAcks(ctx, ids)
-		for _, mec := range mecs {
-			mec.errChan <- err
-		}
+		// TODO(#695): Do something sensible if SendAcks returns an error.
```

Comment:
As Russ suggested, we should probably return an error from Receive. (In a separate PR.)

Reply:
I'm not so sure about that. Anyway, let's move this discussion to #695.